src/main/kotlin/com/google/container/tools/diagnostics/GoogleFeedbackErrorReporter.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.diagnostics
+
+import com.google.common.annotations.VisibleForTesting
+import com.google.container.tools.core.PluginInfo
+import com.intellij.diagnostic.ReportMessages
+import com.intellij.ide.DataManager
+import com.intellij.ide.plugins.PluginManager
+import com.intellij.idea.IdeaLogger
+import com.intellij.notification.NotificationListener
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.application.Application
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ApplicationNamesInfo
+import com.intellij.openapi.application.ex.ApplicationInfoEx
+import com.intellij.openapi.diagnostic.ErrorReportSubmitter
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent
+import com.intellij.openapi.diagnostic.SubmittedReportInfo
+import com.intellij.openapi.extensions.PluginId
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.util.Consumer
+import com.intellij.util.ExceptionUtil
+import java.awt.Component
+
+/**
+ * This class hooks into IntelliJ's error reporting framework. It's based off of
+ * [ErrorReporter.java](https://android.googlesource.com/platform/tools/adt/idea/+/studio-master-dev/android/src/com/android/tools/idea/diagnostics/error/ErrorReporter.java)
+ * in Android Studio.
+ */
+class GoogleFeedbackErrorReporter : ErrorReportSubmitter() {
+
+    override fun getReportActionText(): String {
+        return ErrorReporterBundle.message("error.googlefeedback.message")
+    }
+
+    override fun submit(
+        events: Array<IdeaLoggingEvent>,
+        additionalInfo: String?,
+        parentComponent: Component,
+        consumer: Consumer<SubmittedReportInfo>
+    ): Boolean {
+        return doSubmit(
+            events[0],
```

Comment:
Is this always true? Should we document why a bit more here?

Reply:
What do you think about this array and always accessing just [0]? Is it perfectly safe?

Reply:
I'll look into why this was this way.

Reply:
Yeah. Looks like this is safe. The platform code that calls this, initializes the array as follows:

```
IdeaLoggingEvent[] events = {new IdeaReportingEvent(message, pair.first, pair.second, cluster.plugin)};
```

Why its using an array here is beyond me.

Reply:
Could we make a small comment this array is always size 0 in case we forget in future? Thanks.

Reply:
done - added it in the form of a javadoc param